automata/pda/pda.py

```diff
@@ -24,6 +47,313 @@ class PDA(Automaton, metaclass=abc.ABCMeta):
+                c,
+                n,
+                label=f"<<b>[<i>#{i}</i>]</b>>",
+                arrowsize=arrow_size,
+                fontsize=font_size,
+                color=color.to_string(hex=True),
+                penwidth="2.5",
+            )
+
+        return graph
+
+    def _create_transitions_for_input_diagram(
+        self,
+        graph: pgv.AGraph,
+        input_path: List[Tuple[PDAConfiguration, PDAConfiguration]],
+        edge_drawn: EdgeDrawnDictT,
+        start_color: coloraide.Color,
+        end_color: coloraide.Color,
+        arrow_size: str,
+        font_size: str,
+    ) -> pgv.AGraph:
+        """
+        Add transitions to show execution of the PDA for the given input string
+        """
+
+        interpolation = coloraide.Color.interpolate(
+            [start_color, end_color], space="srgb"
+        )
+
+        # find all transitions in the finite state machine with traversal.
+        for transition_index, (from_state, to_state) in enumerate(input_path, start=1):
+            color = interpolation(transition_index / len(input_path))
+
+            symbol = self._get_symbol_configuration(from_state, to_state)
+            label = self._get_edge_name(*symbol)
+
+            edge_drawn[from_state.state, to_state.state, symbol] = True
+            graph.add_edge(
+                self._get_state_name(from_state.state),
+                self._get_state_name(to_state.state),
+                label=f"<{label} <b>[<i>#{transition_index}</i>]</b>>",
+                arrowsize=arrow_size,
+                fontsize=font_size,
+                color=color.to_string(hex=True),
+                penwidth="2.5",
+            )
+
+        return graph
+
+    def _add_transitions_diagram(
+        self,
+        graph: pgv.AGraph,
+        is_edge_drawn: EdgeDrawnDictT,
+        arrow_size: str,
+        font_size: str,
+    ) -> pgv.AGraph:
+        """
+        Add transitions to between states
+        """
+
+        edge_labels = defaultdict(list)
+        for from_state, to_state, symbol in self.iter_transitions():
+            if is_edge_drawn[from_state, to_state, symbol]:
+                continue
+
+            from_node = self._get_state_name(from_state)
+            to_node = self._get_state_name(to_state)
+            label = self._get_edge_name(*symbol)
+            edge_labels[from_node, to_node].append(label)
+
+        for (from_node, to_node), labels in edge_labels.items():
+            graph.add_edge(
+                from_node,
+                to_node,
+                label="\n".join(sorted(labels)),
+                arrowsize=arrow_size,
+                fontsize=font_size,
+            )
+
+        return graph
+
+    def _add_states_diagram(
+        self,
+        graph: pgv.AGraph,
+        arrow_size: str,
+        font_size: str,
+    ) -> pgv.AGraph:
+        """
+        Add all the states of the PDA
+        """
+
+        # we use a random uuid to make sure that the null node has a
```

Comment:
Could this UUID generation be moved to a helper function? this exact line with comment appears in a few places.